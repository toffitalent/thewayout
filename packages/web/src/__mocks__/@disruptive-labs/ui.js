const actual = jest.requireActual('@disruptive-labs/ui');

const showToast = jest.fn();

actual.Toast.show = showToast;
actual.Toast.danger = showToast;
actual.Toast.info = showToast;
actual.Toast.success = showToast;
actual.Toast.warning = showToast;

module.exports = {
  ...actual,
  showToast,
};
