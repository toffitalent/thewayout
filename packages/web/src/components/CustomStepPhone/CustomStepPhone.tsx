import { Button, Flex, useForm, useWizardContext } from '@disruptive-labs/ui';
import { CustomStepTextField } from '@app/types';
import { formatPhoneNumber } from '@app/utils';
import { PhoneNumberInput } from '../PhoneNumberInput';

export const CustomStepPhone = (props: CustomStepTextField) => {
  const { data: wizardData, next } = useWizardContext<{ phone: string }>();

  const {
    submitForm,
    setValue,
    control,
    formState: { isValid },
  } = useForm({
    mode: 'onChange',
    defaultValues: {
      phone: wizardData.phone ? formatPhoneNumber(wizardData.phone) : '',
    },
    onSubmit: async (data) => {
      await next({
        ...wizardData,
        phone: data.phone ? data.phone.replace(/[^0-9]/g, '') : undefined,
      });
    },
  });

  const { fields } = props;
  if (!fields) {
    return null;
  }
  const [phone] = fields;

  return (
    <Flex maxWidth={96} ml="auto" mr="auto">
      <form onSubmit={submitForm}>
        <PhoneNumberInput
          {...phone}
          control={control}
          fluid
          setPhoneValue={(value: string) => {
            setValue('phone', value, {
              shouldDirty: true,
              shouldValidate: value.length === 14 || value.length === 0,
            });
          }}
          isFieldsRequired={phone.validate?.required}
        />
        <Button colorScheme="primary" fluid type="submit" maxWidth={96} mt={8} disabled={!isValid}>
          Save & Next
        </Button>
      </form>
    </Flex>
  );
};
