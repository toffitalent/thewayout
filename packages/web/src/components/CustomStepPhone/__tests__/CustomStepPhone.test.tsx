import { useWizardContext } from '@disruptive-labs/ui';
import { render, waitFor } from '@test';
import { CustomStepPhone } from '../CustomStepPhone';

jest.mock('@disruptive-labs/ui', () => ({
  ...jest.requireActual('@disruptive-labs/ui'),
  useWizardContext: jest.fn(),
}));

describe('CustomStepPhone', () => {
  beforeEach(() => {
    (useWizardContext as any).mockImplementation(() => ({
      data: {},
      setData: jest.fn(),
      goTo: jest.fn(),
    }));
  });

  test('renders correctly', async () => {
    const { container, findByRole } = render(
      <CustomStepPhone
        fields={[{ name: 'phone', type: 'text', label: 'Phone', placeholder: 'Phone' }]}
      />,
    );

    await waitFor(async () =>
      expect(await findByRole('button', { name: 'Save & Next' })).toBeInTheDocument(),
    );
    expect(container).toMatchSnapshot();
  });
});
