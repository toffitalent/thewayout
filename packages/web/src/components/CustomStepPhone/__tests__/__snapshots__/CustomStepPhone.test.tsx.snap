// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomStepPhone renders correctly 1`] = `
<div>
  <div
    class="maxWidth-96 marginRight-auto marginLeft-auto"
  >
    <form>
      <div
        class="control"
        role="group"
      >
        <label
          class="label"
          for=":r0:"
        >
          Phone
        </label>
        <div
          class="root primary md fluid md"
        >
          <input
            class="input"
            id=":r0:"
            maxlength="14"
            name="phone"
            placeholder="Phone"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="button md solid c-primary fluid maxWidth-96 marginTop-8"
        type="submit"
      >
        Save & Next
      </button>
    </form>
  </div>
</div>
`;
